src/api/eventTypes.js

```diff
@@ -173,3 +175,236 @@ export type RestartEvent = $ReadOnly<{|
+  notifications_stream_id: $PropertyType<InitialDataRealm, 'realm_notifications_stream_id'>,
+  password_auth_enabled: $PropertyType<InitialDataRealm, 'realm_password_auth_enabled'>,
+  plan_type: $PropertyType<InitialDataRealm, 'realm_plan_type'>,
+  presence_disabled: $PropertyType<InitialDataRealm, 'realm_presence_disabled'>,
+  private_message_policy: $PropertyType<InitialDataRealm, 'realm_private_message_policy'>,
+  push_notifications_enabled: $PropertyType<InitialDataRealm, 'realm_push_notifications_enabled'>,
+
+  // TODO: Do servers send this, here or in InitialDataRealm?
+  // realm_restricted_to_domain
+
+  send_welcome_emails: $PropertyType<InitialDataRealm, 'realm_send_welcome_emails'>,
+
+  // TODO: Do servers send this, here or in InitialDataRealm?
+  // realm_show_digest_email
+
+  signup_notifications_stream_id: $PropertyType<
+    InitialDataRealm,
+    'realm_signup_notifications_stream_id',
+  >,
+
+  // TODO(server-5.0): See notes in InitialDataRealm.
+  upload_quota: $PropertyType<InitialDataRealm, 'realm_upload_quota'>,
+  // realm_upload_quota_mib
+
+  user_group_edit_policy: $PropertyType<InitialDataRealm, 'realm_user_group_edit_policy'>,
+  realm_uri: $PropertyType<InitialDataRealm, 'realm_uri'>,
+  video_chat_provider: $PropertyType<InitialDataRealm, 'realm_video_chat_provider'>,
+  waiting_period_threshold: $PropertyType<InitialDataRealm, 'realm_waiting_period_threshold'>,
+
+  // TODO(server-4.0): See note in InitialDataRealm.
+  wildcard_mention_policy: $PropertyType<InitialDataRealm, 'realm_wildcard_mention_policy'>,
+
+  server_avatar_changes_disabled: $PropertyType<InitialDataRealm, 'server_avatar_changes_disabled'>,
+  server_name_changes_disabled: $PropertyType<InitialDataRealm, 'server_name_changes_disabled'>,
+
+  // TODO(server-5.0): See note in InitialDataRealm.
+  // server_needs_upgrade
+
+  server_inline_url_embed_preview: $PropertyType<
+    InitialDataRealm,
+    'server_inline_url_embed_preview',
+  >,
+  server_inline_image_preview: $PropertyType<InitialDataRealm, 'server_inline_image_preview'>,
+  server_generation: $PropertyType<InitialDataRealm, 'server_generation'>,
+  settings_send_digest_emails: $PropertyType<InitialDataRealm, 'settings_send_digest_emails'>,
+  upgrade_text_for_wide_organization_logo: $PropertyType<
+    InitialDataRealm,
+    'upgrade_text_for_wide_organization_logo',
+  >,
+  zulip_plan_is_not_limited: $PropertyType<InitialDataRealm, 'zulip_plan_is_not_limited'>,
+
+  //
+  // Keep alphabetical by the InitialDataRealm property. So by
+  // realm_allow_edit_history, not by allow_edit_history, etc.
+  //
+
+  ...
+}>;
+
+// https://zulip.com/api/get-events#realm-update
+export type RealmUpdateEvent = $ReadOnly<{|
+  ...EventCommon,
+  type: typeof EventTypes.realm,
+
+  op: 'update',
+
+  // TODO(flow): `property` and `value` should correspond
+  property: $Keys<RealmDataForUpdate>,
+  value: $ElementType<$Rest<RealmDataForUpdate, { ... }>, $Keys<RealmDataForUpdate>>,
```

Comment:
I think the way to do this TODO would be to parameterize the type over its `property` type. Like:
```js
export type RealmUpdateEvent<P: $Keys<RealmDataForUpdate>> = // …
  property: P,
  value: $ElementType<RealmDataForUpdate, P>,
  // …
```

But we can look into that after merge.

Reply:
Cool, yeah. That occurred to me, and I tried it, but I ran into an issue with some code for handling the events. Let's debug together? After merge (or before) is fine.

Reply:
Sounds good!